src/Doctrine/Orm/Filter/BackedEnumFilter.php

```diff
@@ -0,0 +1,178 @@
+ *     </services>
+ * </container>
+ * <!-- api/config/api_platform/resources.xml -->
+ * <resources
+ *         xmlns="https://api-platform.com/schema/metadata/resources-3.0"
+ *         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+ *         xsi:schemaLocation="https://api-platform.com/schema/metadata/resources-3.0
+ *         https://api-platform.com/schema/metadata/resources-3.0.xsd">
+ *     <resource class="App\Entity\Book">
+ *         <operations>
+ *             <operation class="ApiPlatform\Metadata\GetCollection">
+ *                 <filters>
+ *                     <filter>book.backed_enum_filter</filter>
+ *                 </filters>
+ *             </operation>
+ *         </operations>
+ *     </resource>
+ * </resources>
+ * ```
+ *
+ * </div>
+ *
+ * Given that the collection endpoint is `/books`, you can filter books with the following query: `/books?status=published`.
+ *
+ * @author Rémi Marseille <marseille.remi@gmail.com>
+ */
+final class BackedEnumFilter extends AbstractFilter
+{
+    use BackedEnumFilterTrait;
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function filterProperty(string $property, mixed $value, QueryBuilder $queryBuilder, QueryNameGeneratorInterface $queryNameGenerator, string $resourceClass, ?Operation $operation = null, array $context = []): void
+    {
+        if (
+            !$this->isPropertyEnabled($property, $resourceClass)
+            || !$this->isPropertyMapped($property, $resourceClass)
+            || !$this->isBackedEnumField($property, $resourceClass)
+        ) {
+            return;
+        }
+
+        $value = $this->normalizeValue($value, $property);
+        if (null === $value) {
+            return;
+        }
+
+        $alias = $queryBuilder->getRootAliases()[0];
+        $field = $property;
+
+        if ($this->isPropertyNested($property, $resourceClass)) {
+            [$alias, $field] = $this->addJoinsForNestedProperty($property, $alias, $queryBuilder, $queryNameGenerator, $resourceClass, Join::INNER_JOIN);
+        }
+
+        $valueParameter = $queryNameGenerator->generateParameterName($field);
+
+        $queryBuilder
+            ->andWhere(\sprintf('%s.%s = :%s', $alias, $field, $valueParameter))
+            ->setParameter($valueParameter, $value);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function isBackedEnumField(string $property, string $resourceClass): bool
+    {
+        $propertyParts = $this->splitPropertyParts($property, $resourceClass);
+        $metadata = $this->getNestedMetadata($resourceClass, $propertyParts['associations']);
+
+        if (!$metadata instanceof ClassMetadata) {
+            return false;
+        }
+
+        $fieldMapping = $metadata->fieldMappings[$propertyParts['field']];
+
+        // Doctrine ORM 2.x returns an array and Doctrine ORM 3.x returns a FieldMapping object
+        if ($fieldMapping instanceof FieldMapping) {
+            $fieldMapping = (array) $fieldMapping;
```

Comment:
does `main` support ORM 2.x ? I think that we can add the code only for `3.x`

Reply:
looks like it is https://github.com/api-platform/core/blob/main/composer.json#L68